// @wix/sdk-runtime/build/transformations/index.js
export const audio = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/audio',
        name: 'transformSDKAudioToRESTAudio',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/audio',
        name: 'transformRESTAudioToSDKAudio',
    },
};
export const address = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/address',
        name: 'transformSDKAddressToRESTAddress',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/address',
        name: 'transformRESTAddressToSDKAddress',
    },
};
export const image = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/image',
        name: 'transformSDKImageToRESTImage',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/image',
        name: 'transformRESTImageToSDKImage',
    },
};
export const videoV2 = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/video-v2',
        name: 'transformSDKVideoV2ToRESTVideoV2',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/video-v2',
        name: 'transformRESTVideoV2ToSDKVideoV2',
    },
};
export const video = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/video',
        name: 'transformSDKVideoToRESTVideo',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/video',
        name: 'transformRESTVideoToSDKVideo',
    },
};
export const document = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/document',
        name: 'transformSDKDocumentToRESTDocument',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/document',
        name: 'transformRESTDocumentToSDKDocument',
    },
};
export const pageUrl = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/page-url',
        name: 'transformSDKPageURLToRESTPageURL',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/page-url',
        name: 'transformRESTPageURLToSDKPageURL',
    },
};
export const pageUrlV2 = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/page-url-v2',
        name: 'transformSDKPageURLV2ToRESTPageURLV2',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/page-url-v2',
        name: 'transformRESTPageURLV2ToSDKPageURLV2',
    },
};
export const float = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/float',
        name: 'transformSDKFloatToRESTFloat',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/float',
        name: 'transformRESTFloatToSDKFloat',
    },
};
export const bytes = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/bytes',
        name: 'transformSDKBytesToRESTBytes',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/bytes',
        name: 'transformRESTBytesToSDKBytes',
    },
};
export const duration = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/duration',
        name: 'transformSDKDurationToRESTDuration',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/duration',
        name: 'transformRESTDurationToSDKDuration',
    },
};
export const timestamp = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/timestamp',
        name: 'transformSDKTimestampToRESTTimestamp',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/timestamp',
        name: 'transformRESTTimestampToSDKTimestamp',
    },
};
export const fieldMask = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/field-mask',
        name: 'transformSDKFieldMaskToRESTFieldMask',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/field-mask',
        name: 'transformRESTFieldMaskToSDKFieldMask',
    },
};
