// @wix/identity_recovery/build/es/src/iam-recovery-v1-recovery-recovery.meta.js
import * as ambassadorWixIamRecoveryV1Recovery from './iam-recovery-v1-recovery-recovery.http';
export function sendRecoveryEmail() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamRecoveryV1Recovery.sendRecoveryEmail(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/send-email',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function sendActivationEmail() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamRecoveryV1Recovery.sendActivationEmail(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/activation-email',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function recover() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamRecoveryV1Recovery.recover(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/recover',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
