// @wix/pricing-plans_plans/build/es/src/pricing-plans-v2-plan-plans.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { archivePlan as universalArchivePlan, clearPrimary as universalClearPrimary, createPlan as universalCreatePlan, getPlan as universalGetPlan, getPlanStats as universalGetPlanStats, listPlans as universalListPlans, listPublicPlans as universalListPublicPlans, makePlanPrimary as universalMakePlanPrimary, queryPublicPlans as universalQueryPublicPlans, setPlanVisibility as universalSetPlanVisibility, updatePlan as universalUpdatePlan, } from './pricing-plans-v2-plan-plans.universal';
export const __metadata = { PACKAGE_NAME: '@wix/pricing-plans' };
export function listPublicPlans(httpClient) {
    return (options) => universalListPublicPlans(options, 
    // @ts-ignore
    { httpClient });
}
export function queryPublicPlans(httpClient) {
    return () => universalQueryPublicPlans(
    // @ts-ignore
    { httpClient });
}
export function getPlan(httpClient) {
    return (_id) => universalGetPlan(_id, 
    // @ts-ignore
    { httpClient });
}
export function listPlans(httpClient) {
    return (options) => universalListPlans(options, 
    // @ts-ignore
    { httpClient });
}
export function getPlanStats(httpClient) {
    return () => universalGetPlanStats(
    // @ts-ignore
    { httpClient });
}
export function createPlan(httpClient) {
    return (plan) => universalCreatePlan(plan, 
    // @ts-ignore
    { httpClient });
}
export function updatePlan(httpClient) {
    return (_id, plan) => universalUpdatePlan(_id, plan, 
    // @ts-ignore
    { httpClient });
}
export function setPlanVisibility(httpClient) {
    return (_id, visible) => universalSetPlanVisibility(_id, visible, 
    // @ts-ignore
    { httpClient });
}
export function makePlanPrimary(httpClient) {
    return (_id) => universalMakePlanPrimary(_id, 
    // @ts-ignore
    { httpClient });
}
export function clearPrimary(httpClient) {
    return () => universalClearPrimary(
    // @ts-ignore
    { httpClient });
}
export function archivePlan(httpClient) {
    return (_id) => universalArchivePlan(_id, 
    // @ts-ignore
    { httpClient });
}
export const onPlanUpdated = EventDefinition('wix.pricing_plans.plan_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onPlanCreated = EventDefinition('wix.pricing_plans.plan_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onPlanBuyerCanCancelUpdated = EventDefinition('wix.pricing_plans.plan_buyer_can_cancel_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.plan.createdDate' },
            { path: 'data.plan.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onPlanArchived = EventDefinition('wix.pricing_plans.plan_plan_archived', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.plan.createdDate' },
            { path: 'data.plan.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export { AppliedAt, ArchivedFilter, PeriodUnit, PublicFilter, SortOrder, WebhookIdentityType, } from './pricing-plans-v2-plan-plans.universal';
