// @wix/pricing-plans_plans/build/es/src/pricing-plans-v2-plan-plans.meta.js
import * as ambassadorWixPricingPlansV2Plan from './pricing-plans-v2-plan-plans.http';
export function listPublicPlans() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.listPublicPlans(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/plans/public',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryPublicPlans() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.queryPublicPlans(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/plans/public/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getPlan() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.getPlan(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/plans/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listPlans() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.listPlans(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/plans',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getPlanStats() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.getPlanStats(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/plans/stats',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createPlan() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.createPlan(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/plans',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updatePlan() {
    const payload = { plan: { id: ':planId' } };
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.updatePlan(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/plans/{plan.id}',
        pathParams: { planId: 'planId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setPlanVisibility() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.setPlanVisibility(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v2/plans/{id}/visibility',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function makePlanPrimary() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.makePlanPrimary(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/plans/{id}/make-primary',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function clearPrimary() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.clearPrimary(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/plans/clear-primary',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function archivePlan() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.archivePlan(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/plans/{id}/archive',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
