// @wix/redirects_redirects/build/es/src/headless-v1-redirect-session-redirects.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { createRedirectSession as universalCreateRedirectSession, } from './headless-v1-redirect-session-redirects.universal';
export const __metadata = { PACKAGE_NAME: '@wix/redirects' };
export function createRedirectSession(httpClient) {
    return (options) => universalCreateRedirectSession(options, 
    // @ts-ignore
    { httpClient });
}
export const onRedirectSessionCreated = EventDefinition('wix.headless.v1.redirect_session_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'metadata.eventTime' }],
    },
])))();
export { AttachPagesResponseStatus, CallbackType, LocationType, MembersAccountSection, Prompt, Status, WebhookIdentityType, } from './headless-v1-redirect-session-redirects.universal';
