// @wix/redirects_redirects/build/es/src/headless-v1-redirect-session-redirects.types.js
export var LocationType;
(function (LocationType) {
    LocationType["UNDEFINED"] = "UNDEFINED";
    LocationType["OWNER_BUSINESS"] = "OWNER_BUSINESS";
    LocationType["OWNER_CUSTOM"] = "OWNER_CUSTOM";
    LocationType["CUSTOM"] = "CUSTOM";
})(LocationType || (LocationType = {}));
/** Currently only `none` and `login` are supported. */
export var Prompt;
(function (Prompt) {
    Prompt["login"] = "login";
    Prompt["none"] = "none";
    Prompt["consent"] = "consent";
    Prompt["select_account"] = "select_account";
})(Prompt || (Prompt = {}));
export var MembersAccountSection;
(function (MembersAccountSection) {
    /** default - account info section in "my account" */
    MembersAccountSection["ACCOUNT_INFO"] = "ACCOUNT_INFO";
    /** My Bookings section in "my account" */
    MembersAccountSection["BOOKINGS"] = "BOOKINGS";
    /** My Orders section in "my account" */
    MembersAccountSection["ORDERS"] = "ORDERS";
    /** Subscriptions section in "my account" */
    MembersAccountSection["SUBSCRIPTIONS"] = "SUBSCRIPTIONS";
    /** Events section in "my account" */
    MembersAccountSection["EVENTS"] = "EVENTS";
})(MembersAccountSection || (MembersAccountSection = {}));
export var AttachPagesResponseStatus;
(function (AttachPagesResponseStatus) {
    /** Invalid value - default */
    AttachPagesResponseStatus["UNKNOWN"] = "UNKNOWN";
    /** Pages were successfully attached */
    AttachPagesResponseStatus["SUCCESS"] = "SUCCESS";
    /** No pages were attached because the site is already published */
    AttachPagesResponseStatus["NO_ACTION"] = "NO_ACTION";
    /** An error occurred, such as site or app not being found */
    AttachPagesResponseStatus["ERROR"] = "ERROR";
})(AttachPagesResponseStatus || (AttachPagesResponseStatus = {}));
export var CallbackType;
(function (CallbackType) {
    /** Invalid value - default */
    CallbackType["UNKNOWN"] = "UNKNOWN";
    /** The callback URL is used for the logout flow */
    CallbackType["LOGOUT"] = "LOGOUT";
    /** The callback URL is used for a checkout flow */
    CallbackType["CHECKOUT"] = "CHECKOUT";
    /** The callback URL is used for the authorize flow */
    CallbackType["AUTHORIZE"] = "AUTHORIZE";
})(CallbackType || (CallbackType = {}));
export var Status;
(function (Status) {
    /** Invalid value - default */
    Status["UNKNOWN"] = "UNKNOWN";
    /** templates were successfully attached */
    Status["SUCCESS"] = "SUCCESS";
    /** An error occurred */
    Status["ERROR"] = "ERROR";
})(Status || (Status = {}));
export var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));
