// @wix/pricing-plans_orders/build/cjs/src/pricing-plans-v2-order-orders.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Status = exports.SortOrder = exports.Set = exports.ReasonNotSuspendable = exports.PeriodUnit = exports.PaymentStatus = exports.OrderType = exports.OrderStatus = exports.OrderMethod = exports.CancellationEffectiveAt = exports.CancellationCause = exports.onOrderResumed = exports.onOrderPaused = exports.onOrderMarkedAsPaid = exports.onOrderEndDatePostponed = exports.onOrderEnded = exports.onOrderAutoRenewCanceled = exports.onOrderCycleStarted = exports.onOrderStarted = exports.onOrderPurchased = exports.onOrderStartDateChanged = exports.onOrderUpdated = exports.onOrderCreated = exports.onOrderCanceled = exports.resumeOrder = exports.pauseOrder = exports.markAsPaid = exports.cancelOrder = exports.postponeEndDate = exports.managementListOrders = exports.managementGetOrder = exports.getPricePreview = exports.getOfflineOrderPreview = exports.createOfflineOrder = exports.requestCancellation = exports.memberListOrders = exports.memberGetOrder = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const pricing_plans_v2_order_orders_universal_1 = require("./pricing-plans-v2-order-orders.universal");
exports.__metadata = { PACKAGE_NAME: '@wix/pricing-plans' };
function memberGetOrder(httpClient) {
    return (_id, options) => (0, pricing_plans_v2_order_orders_universal_1.memberGetOrder)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.memberGetOrder = memberGetOrder;
function memberListOrders(httpClient) {
    return (options) => (0, pricing_plans_v2_order_orders_universal_1.memberListOrders)(options, 
    // @ts-ignore
    { httpClient });
}
exports.memberListOrders = memberListOrders;
function requestCancellation(httpClient) {
    return (_id, effectiveAt) => (0, pricing_plans_v2_order_orders_universal_1.requestCancellation)(_id, effectiveAt, 
    // @ts-ignore
    { httpClient });
}
exports.requestCancellation = requestCancellation;
function createOfflineOrder(httpClient) {
    return (planId, memberId, options) => (0, pricing_plans_v2_order_orders_universal_1.createOfflineOrder)(planId, memberId, options, 
    // @ts-ignore
    { httpClient });
}
exports.createOfflineOrder = createOfflineOrder;
function getOfflineOrderPreview(httpClient) {
    return (planId, memberId, options) => (0, pricing_plans_v2_order_orders_universal_1.getOfflineOrderPreview)(planId, memberId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getOfflineOrderPreview = getOfflineOrderPreview;
function getPricePreview(httpClient) {
    return (planId, options) => (0, pricing_plans_v2_order_orders_universal_1.getPricePreview)(planId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getPricePreview = getPricePreview;
function managementGetOrder(httpClient) {
    return (_id, options) => (0, pricing_plans_v2_order_orders_universal_1.managementGetOrder)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.managementGetOrder = managementGetOrder;
function managementListOrders(httpClient) {
    return (options) => (0, pricing_plans_v2_order_orders_universal_1.managementListOrders)(options, 
    // @ts-ignore
    { httpClient });
}
exports.managementListOrders = managementListOrders;
function postponeEndDate(httpClient) {
    return (_id, endDate) => (0, pricing_plans_v2_order_orders_universal_1.postponeEndDate)(_id, endDate, 
    // @ts-ignore
    { httpClient });
}
exports.postponeEndDate = postponeEndDate;
function cancelOrder(httpClient) {
    return (_id, effectiveAt) => (0, pricing_plans_v2_order_orders_universal_1.cancelOrder)(_id, effectiveAt, 
    // @ts-ignore
    { httpClient });
}
exports.cancelOrder = cancelOrder;
function markAsPaid(httpClient) {
    return (_id) => (0, pricing_plans_v2_order_orders_universal_1.markAsPaid)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.markAsPaid = markAsPaid;
function pauseOrder(httpClient) {
    return (_id) => (0, pricing_plans_v2_order_orders_universal_1.pauseOrder)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.pauseOrder = pauseOrder;
function resumeOrder(httpClient) {
    return (_id) => (0, pricing_plans_v2_order_orders_universal_1.resumeOrder)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.resumeOrder = resumeOrder;
exports.onOrderCanceled = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_canceled', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderCreated = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.startDate' },
            { path: 'entity.endDate' },
            { path: 'entity.earliestEndDate' },
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.cancellation.requestedDate' },
            { path: 'entity.pausePeriods.pauseDate' },
            { path: 'entity.pausePeriods.resumeDate' },
            { path: 'entity.currentCycle.startedDate' },
            { path: 'entity.currentCycle.endedDate' },
            { path: 'entity.cycles.startedDate' },
            { path: 'entity.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderUpdated = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.startDate' },
            { path: 'entity.endDate' },
            { path: 'entity.earliestEndDate' },
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.cancellation.requestedDate' },
            { path: 'entity.pausePeriods.pauseDate' },
            { path: 'entity.pausePeriods.resumeDate' },
            { path: 'entity.currentCycle.startedDate' },
            { path: 'entity.currentCycle.endedDate' },
            { path: 'entity.cycles.startedDate' },
            { path: 'entity.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderStartDateChanged = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_start_date_changed', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderPurchased = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_purchased', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderStarted = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_started', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderCycleStarted = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_cycle_started', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderAutoRenewCanceled = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_auto_renew_canceled', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderEnded = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_ended', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderEndDatePostponed = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_end_date_postponed', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderMarkedAsPaid = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_marked_as_paid', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderPaused = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_paused', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onOrderResumed = (0, sdk_types_1.EventDefinition)('wix.pricing_plans.v2.order_resumed', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var pricing_plans_v2_order_orders_universal_2 = require("./pricing-plans-v2-order-orders.universal");
Object.defineProperty(exports, "CancellationCause", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.CancellationCause; } });
Object.defineProperty(exports, "CancellationEffectiveAt", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.CancellationEffectiveAt; } });
Object.defineProperty(exports, "OrderMethod", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.OrderMethod; } });
Object.defineProperty(exports, "OrderStatus", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.OrderStatus; } });
Object.defineProperty(exports, "OrderType", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.OrderType; } });
Object.defineProperty(exports, "PaymentStatus", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.PaymentStatus; } });
Object.defineProperty(exports, "PeriodUnit", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.PeriodUnit; } });
Object.defineProperty(exports, "ReasonNotSuspendable", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.ReasonNotSuspendable; } });
Object.defineProperty(exports, "Set", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.Set; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.SortOrder; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.Status; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return pricing_plans_v2_order_orders_universal_2.WebhookIdentityType; } });
