// @wix/pricing-plans_orders/build/cjs/src/pricing-plans-v2-order-orders.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.resumeOrder = exports.pauseOrder = exports.markAsPaid = exports.cancelOrder = exports.postponeEndDate = exports.managementListOrders = exports.managementGetOrder = exports.getPricePreview = exports.getOfflineOrderPreview = exports.createOfflineOrder = exports.requestCancellation = exports.memberListOrders = exports.memberGetOrder = void 0;
const ambassadorWixPricingPlansV2Order = __importStar(require("./pricing-plans-v2-order-orders.http"));
function memberGetOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.memberGetOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/member/orders/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.memberGetOrder = memberGetOrder;
function memberListOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Order.memberListOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/member/orders',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.memberListOrders = memberListOrders;
function requestCancellation() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.requestCancellation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/member/orders/{id}/cancel',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.requestCancellation = requestCancellation;
function createOfflineOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Order.createOfflineOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/checkout/orders/offline',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createOfflineOrder = createOfflineOrder;
function getOfflineOrderPreview() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Order.getOfflineOrderPreview(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/checkout/orders/preview-offline',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getOfflineOrderPreview = getOfflineOrderPreview;
function getPricePreview() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Order.getPricePreview(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/checkout/orders/price-preview',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getPricePreview = getPricePreview;
function managementGetOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.managementGetOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/orders/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.managementGetOrder = managementGetOrder;
function managementListOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Order.managementListOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/orders',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.managementListOrders = managementListOrders;
function postponeEndDate() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.postponeEndDate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/orders/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.postponeEndDate = postponeEndDate;
function cancelOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.cancelOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/orders/{id}/cancel',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.cancelOrder = cancelOrder;
function markAsPaid() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.markAsPaid(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/orders/{id}/mark-as-paid',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.markAsPaid = markAsPaid;
function pauseOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.pauseOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/orders/{id}/pause',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.pauseOrder = pauseOrder;
function resumeOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.resumeOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/orders/{id}/resume',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.resumeOrder = resumeOrder;
