// @wix/redirects_redirects/build/cjs/src/headless-v1-redirect-session-redirects.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Status = exports.Prompt = exports.MembersAccountSection = exports.LocationType = exports.CallbackType = exports.AttachPagesResponseStatus = exports.onRedirectSessionCreated = exports.createRedirectSession = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const headless_v1_redirect_session_redirects_universal_1 = require("./headless-v1-redirect-session-redirects.universal");
exports.__metadata = { PACKAGE_NAME: '@wix/redirects' };
function createRedirectSession(httpClient) {
    return (options) => (0, headless_v1_redirect_session_redirects_universal_1.createRedirectSession)(options, 
    // @ts-ignore
    { httpClient });
}
exports.createRedirectSession = createRedirectSession;
exports.onRedirectSessionCreated = (0, sdk_types_1.EventDefinition)('wix.headless.v1.redirect_session_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'metadata.eventTime' }],
    },
])))();
var headless_v1_redirect_session_redirects_universal_2 = require("./headless-v1-redirect-session-redirects.universal");
Object.defineProperty(exports, "AttachPagesResponseStatus", { enumerable: true, get: function () { return headless_v1_redirect_session_redirects_universal_2.AttachPagesResponseStatus; } });
Object.defineProperty(exports, "CallbackType", { enumerable: true, get: function () { return headless_v1_redirect_session_redirects_universal_2.CallbackType; } });
Object.defineProperty(exports, "LocationType", { enumerable: true, get: function () { return headless_v1_redirect_session_redirects_universal_2.LocationType; } });
Object.defineProperty(exports, "MembersAccountSection", { enumerable: true, get: function () { return headless_v1_redirect_session_redirects_universal_2.MembersAccountSection; } });
Object.defineProperty(exports, "Prompt", { enumerable: true, get: function () { return headless_v1_redirect_session_redirects_universal_2.Prompt; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return headless_v1_redirect_session_redirects_universal_2.Status; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return headless_v1_redirect_session_redirects_universal_2.WebhookIdentityType; } });
