// @wix/pricing-plans_plans/build/cjs/src/pricing-plans-v2-plan-plans.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.archivePlan = exports.clearPrimary = exports.makePlanPrimary = exports.setPlanVisibility = exports.updatePlan = exports.createPlan = exports.getPlanStats = exports.listPlans = exports.getPlan = exports.queryPublicPlans = exports.listPublicPlans = void 0;
const ambassadorWixPricingPlansV2Plan = __importStar(require("./pricing-plans-v2-plan-plans.http"));
function listPublicPlans() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.listPublicPlans(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/plans/public',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listPublicPlans = listPublicPlans;
function queryPublicPlans() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.queryPublicPlans(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/plans/public/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryPublicPlans = queryPublicPlans;
function getPlan() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.getPlan(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/plans/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getPlan = getPlan;
function listPlans() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.listPlans(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/plans',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listPlans = listPlans;
function getPlanStats() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.getPlanStats(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/plans/stats',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getPlanStats = getPlanStats;
function createPlan() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.createPlan(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/plans',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createPlan = createPlan;
function updatePlan() {
    const payload = { plan: { id: ':planId' } };
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.updatePlan(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/plans/{plan.id}',
        pathParams: { planId: 'planId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updatePlan = updatePlan;
function setPlanVisibility() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.setPlanVisibility(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v2/plans/{id}/visibility',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setPlanVisibility = setPlanVisibility;
function makePlanPrimary() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.makePlanPrimary(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/plans/{id}/make-primary',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.makePlanPrimary = makePlanPrimary;
function clearPrimary() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.clearPrimary(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/plans/clear-primary',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.clearPrimary = clearPrimary;
function archivePlan() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Plan.archivePlan(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/plans/{id}/archive',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.archivePlan = archivePlan;
