// @wix/identity_authentication/build/es/src/iam-authentication-v1-authentication-authentication.context.js
import { register as publicRegister, login as publicLogin, registerV2 as publicRegisterV2, loginV2 as publicLoginV2, changePassword as publicChangePassword, loginWithIdpConnectionTokenParams as publicLoginWithIdpConnectionTokenParams, proceedToNextState as publicProceedToNextState, signOn as publicSignOn, logout as publicLogout, } from './iam-authentication-v1-authentication-authentication.public';
export const register = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(publicRegister, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
export const login = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(publicLogin, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
export const registerV2 = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(publicRegisterV2, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
export const loginV2 = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(publicLoginV2, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
export const changePassword = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(publicChangePassword, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
export const loginWithIdpConnectionTokenParams = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(publicLoginWithIdpConnectionTokenParams, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
export const proceedToNextState = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(publicProceedToNextState, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
export const signOn = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(publicSignOn, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
export const logout = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(publicLogout, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
