// @wix/identity_authentication/build/es/src/iam-authentication-v1-authentication-authentication.meta.js
import * as ambassadorWixIamAuthenticationV1Authentication from './iam-authentication-v1-authentication-authentication.http';
export function register() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.register(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/register',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function login() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.login(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/login',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function registerV2() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.registerV2(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/register',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function loginV2() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.loginV2(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/login',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function changePassword() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.changePassword(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/change-password',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function loginWithIdpConnectionTokenParams() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.loginWithIdpConnectionTokenParams(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/sso/login-with-token-params',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function proceedToNextState() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.proceedToNextState(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/state/proceed',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function signOn() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.signOn(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/sign-on',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function logout() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.logout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/logout',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
