// @wix/metro-runtime/dist/cjs/velo-events.js
"use strict";

exports.__esModule = true;
exports.toVeloEvent = toVeloEvent;
var _lodash = require("lodash");
var _veloIndex = require("./velo-index");
exports.RootPath = _veloIndex.RootPath;
function toVeloEvent(domainEvent, serializerParams) {
  const veloEvent = {
    metadata: toMetadata(domainEvent)
  };
  if (shouldContainPayload(domainEvent)) {
    const {
      key,
      payload
    } = toVeloPayload(domainEvent, serializerParams);
    veloEvent[key] = payload;
  }
  return veloEvent;
}
function toVeloPayload(domainEvent, serializerParams) {
  const protoPayload = (0, _veloIndex.extractDomainEventPayload)(domainEvent);
  const payload = (0, _veloIndex.serializer)(serializerParams).fromJSON(protoPayload);
  const key = domainEvent.actionEvent ? 'data' : 'entity';
  return {
    key,
    payload
  };
}
function toMetadata(domainEvent) {
  return (0, _lodash.pick)(domainEvent, 'id', 'entityId', 'eventTime', 'triggeredByAnonymizeRequest');
}
function isEntityIncludedInDeletedEvent(deletedEvent) {
  return deletedEvent.hasOwnProperty('deletedEntityAsJson');
}
function shouldContainPayload(domainEvent) {
  return !domainEvent.deletedEvent || isEntityIncludedInDeletedEvent(domainEvent.deletedEvent);
}
