// @wix/identity_authentication/build/cjs/src/iam-authentication-v1-authentication-authentication.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.logout = exports.signOn = exports.proceedToNextState = exports.loginWithIdpConnectionTokenParams = exports.changePassword = exports.loginV2 = exports.registerV2 = exports.login = exports.register = void 0;
const iam_authentication_v1_authentication_authentication_public_1 = require("./iam-authentication-v1-authentication-authentication.public");
const register = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(iam_authentication_v1_authentication_authentication_public_1.register, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
exports.register = register;
const login = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(iam_authentication_v1_authentication_authentication_public_1.login, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
exports.login = login;
const registerV2 = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(iam_authentication_v1_authentication_authentication_public_1.registerV2, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
exports.registerV2 = registerV2;
const loginV2 = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(iam_authentication_v1_authentication_authentication_public_1.loginV2, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
exports.loginV2 = loginV2;
const changePassword = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(iam_authentication_v1_authentication_authentication_public_1.changePassword, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
exports.changePassword = changePassword;
const loginWithIdpConnectionTokenParams = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(iam_authentication_v1_authentication_authentication_public_1.loginWithIdpConnectionTokenParams, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
exports.loginWithIdpConnectionTokenParams = loginWithIdpConnectionTokenParams;
const proceedToNextState = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(iam_authentication_v1_authentication_authentication_public_1.proceedToNextState, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
exports.proceedToNextState = proceedToNextState;
const signOn = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(iam_authentication_v1_authentication_authentication_public_1.signOn, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
exports.signOn = signOn;
const logout = (...args) => {
    const context = 
    // @ts-expect-error
    typeof $wixContext !== 'undefined'
        ? // @ts-expect-error
            $wixContext
        : // @ts-expect-error
            globalThis.__wix_context__;
    if (!context) {
        throw new Error('Wix context is not available. Make sure to initialize the Wix context before using SDK modules');
    }
    return context
        .initWixModules(iam_authentication_v1_authentication_authentication_public_1.logout, args[1]?.supressAuth ? true : false)
        .apply(undefined, args);
};
exports.logout = logout;
