// @wix/identity_authentication/build/cjs/src/iam-authentication-v1-authentication-authentication.universal.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.logout = exports.signOn = exports.proceedToNextState = exports.loginWithIdpConnectionTokenParams = exports.changePassword = exports.loginV2 = exports.registerV2 = exports.login = exports.register = exports.onAuthenticationSuccessfulLoginEvent = exports.TenantType = exports.StateType = exports.StateStatus = exports.Reason = exports.StatusName = exports.AddressTag = exports.PhoneTag = exports.EmailTag = exports.PrivacyStatus = void 0;
const transform_error_1 = require("@wix/sdk-runtime/transform-error");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const sdk_types_1 = require("@wix/sdk-types");
const ambassadorWixIamAuthenticationV1Authentication = __importStar(require("./iam-authentication-v1-authentication-authentication.http"));
const address_1 = require("@wix/sdk-runtime/transformations/address");
const address_2 = require("@wix/sdk-runtime/transformations/address");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
var PrivacyStatus;
(function (PrivacyStatus) {
    PrivacyStatus["UNDEFINED"] = "UNDEFINED";
    PrivacyStatus["PUBLIC"] = "PUBLIC";
    PrivacyStatus["PRIVATE"] = "PRIVATE";
})(PrivacyStatus || (exports.PrivacyStatus = PrivacyStatus = {}));
var EmailTag;
(function (EmailTag) {
    EmailTag["UNTAGGED"] = "UNTAGGED";
    EmailTag["MAIN"] = "MAIN";
    EmailTag["HOME"] = "HOME";
    EmailTag["WORK"] = "WORK";
})(EmailTag || (exports.EmailTag = EmailTag = {}));
var PhoneTag;
(function (PhoneTag) {
    PhoneTag["UNTAGGED"] = "UNTAGGED";
    PhoneTag["MAIN"] = "MAIN";
    PhoneTag["HOME"] = "HOME";
    PhoneTag["MOBILE"] = "MOBILE";
    PhoneTag["WORK"] = "WORK";
    PhoneTag["FAX"] = "FAX";
})(PhoneTag || (exports.PhoneTag = PhoneTag = {}));
var AddressTag;
(function (AddressTag) {
    AddressTag["UNTAGGED"] = "UNTAGGED";
    AddressTag["HOME"] = "HOME";
    AddressTag["WORK"] = "WORK";
    AddressTag["BILLING"] = "BILLING";
    AddressTag["SHIPPING"] = "SHIPPING";
})(AddressTag || (exports.AddressTag = AddressTag = {}));
var StatusName;
(function (StatusName) {
    StatusName["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
    StatusName["PENDING"] = "PENDING";
    StatusName["ACTIVE"] = "ACTIVE";
    StatusName["DELETED"] = "DELETED";
    StatusName["BLOCKED"] = "BLOCKED";
    StatusName["OFFLINE"] = "OFFLINE";
})(StatusName || (exports.StatusName = StatusName = {}));
var Reason;
(function (Reason) {
    Reason["UNKNOWN_REASON"] = "UNKNOWN_REASON";
    Reason["PENDING_ADMIN_APPROVAL_REQUIRED"] = "PENDING_ADMIN_APPROVAL_REQUIRED";
    Reason["PENDING_EMAIL_VERIFICATION_REQUIRED"] = "PENDING_EMAIL_VERIFICATION_REQUIRED";
})(Reason || (exports.Reason = Reason = {}));
var StateStatus;
(function (StateStatus) {
    StateStatus["DONE"] = "DONE";
    StateStatus["REQUIRE_OWNER_APPROVAL"] = "REQUIRE_OWNER_APPROVAL";
    StateStatus["STATUS_CHECK"] = "STATUS_CHECK";
})(StateStatus || (exports.StateStatus = StateStatus = {}));
var StateType;
(function (StateType) {
    /** Initial unknown state. */
    StateType["UNKNOWN_STATE"] = "UNKNOWN_STATE";
    /** The operation completed successfully. */
    StateType["SUCCESS"] = "SUCCESS";
    /** State that indicates that the member needs owner approval to proceed, available action in: OwnerApprovalStateHandler */
    StateType["REQUIRE_OWNER_APPROVAL"] = "REQUIRE_OWNER_APPROVAL";
    /**
     * State that indicates a member waiting for verification, available action are: verifyDuringAuthentication or resendDuringAuthentication
     * https://dev.wix.com/docs/rest/api-reference/auth-management/verification-v1/verify-during-authentication
     */
    StateType["REQUIRE_EMAIL_VERIFICATION"] = "REQUIRE_EMAIL_VERIFICATION";
    /** State that indicates checking that the status is not one of the `invalidStates` before proceeding. */
    StateType["STATUS_CHECK"] = "STATUS_CHECK";
})(StateType || (exports.StateType = StateType = {}));
var TenantType;
(function (TenantType) {
    TenantType["UNKNOWN_TENANT_TYPE"] = "UNKNOWN_TENANT_TYPE";
    TenantType["ACCOUNT"] = "ACCOUNT";
    TenantType["SITE"] = "SITE";
    TenantType["ROOT"] = "ROOT";
})(TenantType || (exports.TenantType = TenantType = {}));
const _addressWrapper = { address: 'wix.common.Address' };
const _customField = { value: '_v1CustomValue' };
const _identity = { identityProfile: '_identityProfile' };
const _identityProfile = {
    customFields: '_customField',
    addresses: '_addressWrapper',
};
const _successfulLoginEvent = { identity: '_identity' };
const _v1CustomValue = { listValue: '_v1ListValue', mapValue: '_v1MapValue' };
const _v1ListValue = { value: '_v1CustomValue' };
const _v1MapValue = { value: 'Map#_v1CustomValue' };
exports.onAuthenticationSuccessfulLoginEvent = (0, sdk_types_1.EventDefinition)('wix.iam.authentication.v1.authentication_successful_login_event', true, {
    rootSchema: _successfulLoginEvent,
    depSchemas: {
        _addressWrapper,
        _customField,
        _identity,
        _identityProfile,
        _v1CustomValue,
        _v1ListValue,
        _v1MapValue,
    },
})();
/** @public
 * @documentationMaturity preview
 * @requiredField identity
 */
async function register(identity, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = (0, transform_paths_1.transformPaths)((0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        identity: identity,
        authenticatorId: options?.authenticatorId,
        inputs: options?.inputs,
        captchaTokens: options?.captchaTokens,
    }), [
        {
            transformFn: address_1.transformSDKAddressToRESTAddress,
            paths: [{ path: 'identity.identityProfile.addresses.address' }],
        },
    ]);
    const reqOpts = ambassadorWixIamAuthenticationV1Authentication.register(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(result.data, [
            {
                transformFn: address_2.transformRESTAddressToSDKAddress,
                paths: [{ path: 'identity.identityProfile.addresses.address' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                identity: '$[0]',
                authenticatorId: '$[1].authenticatorId',
                inputs: '$[1].inputs',
                captchaTokens: '$[1].captchaTokens',
            },
            singleArgumentUnchanged: false,
        }, ['identity', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.register = register;
/** @public
 * @documentationMaturity preview
 * @requiredField identifier
 */
async function login(identifier, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        identifier: identifier,
        authenticatorId: options?.authenticatorId,
        inputs: options?.inputs,
        captchaTokens: options?.captchaTokens,
        loginId: options?.loginId,
    });
    const reqOpts = ambassadorWixIamAuthenticationV1Authentication.login(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(result.data, [
            {
                transformFn: address_2.transformRESTAddressToSDKAddress,
                paths: [{ path: 'identity.identityProfile.addresses.address' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                identifier: '$[0]',
                authenticatorId: '$[1].authenticatorId',
                inputs: '$[1].inputs',
                captchaTokens: '$[1].captchaTokens',
                loginId: '$[1].loginId',
            },
            singleArgumentUnchanged: false,
        }, ['identifier', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.login = login;
/**
 * Registers a new member.
 *
 * Typically, after a sucessful registration, you generate and use member tokens for the
 * registered member so that subsequent API calls are called as part of a member session.
 *
 * If the email used to register the member already exists as a contact email, the registering
 * member need to verify the email address using a code that is sent to the address.
 * @param loginId - Identifier of registering member.
 * @public
 * @documentationMaturity preview
 * @requiredField loginId
 */
async function registerV2(loginId, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = (0, transform_paths_1.transformPaths)((0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        loginId: loginId,
        password: options?.password,
        profile: options?.profile,
        captchaTokens: options?.captchaTokens,
        clientMetaData: options?.clientMetaData,
    }), [
        {
            transformFn: address_1.transformSDKAddressToRESTAddress,
            paths: [{ path: 'profile.addresses.address' }],
        },
    ]);
    const reqOpts = ambassadorWixIamAuthenticationV1Authentication.registerV2(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(result.data, [
            {
                transformFn: address_2.transformRESTAddressToSDKAddress,
                paths: [{ path: 'identity.identityProfile.addresses.address' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                loginId: '$[0]',
                password: '$[1].password',
                profile: '$[1].profile',
                captchaTokens: '$[1].captchaTokens',
                clientMetaData: '$[1].clientMetaData',
            },
            singleArgumentUnchanged: false,
        }, ['loginId', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.registerV2 = registerV2;
/**
 * Logs in an existing user.
 *
 * Typically, after a sucessful login, you generate and use member tokens for the
 * logged-in member so that subsequent API calls are called as part of a member session.
 * @param loginId - Identifier of member logging in.
 * @public
 * @documentationMaturity preview
 * @requiredField loginId
 */
async function loginV2(loginId, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        loginId: loginId,
        password: options?.password,
        captchaTokens: options?.captchaTokens,
    });
    const reqOpts = ambassadorWixIamAuthenticationV1Authentication.loginV2(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(result.data, [
            {
                transformFn: address_2.transformRESTAddressToSDKAddress,
                paths: [{ path: 'identity.identityProfile.addresses.address' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                loginId: '$[0]',
                password: '$[1].password',
                captchaTokens: '$[1].captchaTokens',
            },
            singleArgumentUnchanged: false,
        }, ['loginId', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.loginV2 = loginV2;
/**
 * Changes the password of a logged in user.
 * @param newPassword - The new password to set for the logged in user
 * @public
 * @documentationMaturity preview
 * @requiredField newPassword
 */
async function changePassword(newPassword) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        newPassword: newPassword,
    });
    const reqOpts = ambassadorWixIamAuthenticationV1Authentication.changePassword(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { newPassword: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['newPassword']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.changePassword = changePassword;
/** @public
 * @documentationMaturity preview
 */
async function loginWithIdpConnectionTokenParams(options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        idpConnectionId: options?.idpConnectionId,
        tokenParams: options?.tokenParams,
    });
    const reqOpts = ambassadorWixIamAuthenticationV1Authentication.loginWithIdpConnectionTokenParams(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(result.data, [
            {
                transformFn: address_2.transformRESTAddressToSDKAddress,
                paths: [{ path: 'identity.identityProfile.addresses.address' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                idpConnectionId: '$[0].idpConnectionId',
                tokenParams: '$[0].tokenParams',
            },
            singleArgumentUnchanged: false,
        }, ['options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.loginWithIdpConnectionTokenParams = loginWithIdpConnectionTokenParams;
/** @public
 * @documentationMaturity preview
 */
async function proceedToNextState(options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        stateToken: options?.stateToken,
    });
    const reqOpts = ambassadorWixIamAuthenticationV1Authentication.proceedToNextState(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(result.data, [
            {
                transformFn: address_2.transformRESTAddressToSDKAddress,
                paths: [{ path: 'identity.identityProfile.addresses.address' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { stateToken: '$[0].stateToken' },
            singleArgumentUnchanged: false,
        }, ['options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.proceedToNextState = proceedToNextState;
/** @param loginId - the identifier of the identity
 * @public
 * @documentationMaturity preview
 * @requiredField loginId
 */
async function signOn(loginId, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = (0, transform_paths_1.transformPaths)((0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        loginId: loginId,
        profile: options?.profile,
        verifyEmail: options?.verifyEmail,
        mergeExistingContact: options?.mergeExistingContact,
    }), [
        {
            transformFn: address_1.transformSDKAddressToRESTAddress,
            paths: [{ path: 'profile.addresses.address' }],
        },
    ]);
    const reqOpts = ambassadorWixIamAuthenticationV1Authentication.signOn(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(result.data, [
            {
                transformFn: address_2.transformRESTAddressToSDKAddress,
                paths: [{ path: 'identity.identityProfile.addresses.address' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                loginId: '$[0]',
                profile: '$[1].profile',
                verifyEmail: '$[1].verifyEmail',
                mergeExistingContact: '$[1].mergeExistingContact',
            },
            singleArgumentUnchanged: false,
        }, ['loginId', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.signOn = signOn;
/**
 * Logs out a member.
 * @public
 * @documentationMaturity preview
 */
async function logout(options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        postLogoutRedirectUri: options?.postLogoutRedirectUri,
        clientId: options?.clientId,
    });
    const reqOpts = ambassadorWixIamAuthenticationV1Authentication.logout(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                postLogoutRedirectUri: '$[0].postLogoutRedirectUri',
                clientId: '$[0].clientId',
            },
            singleArgumentUnchanged: false,
        }, ['options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.logout = logout;
