// @wix/pricing-plans_orders/build/es/src/pricing-plans-v2-order-orders.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { cancelOrder as universalCancelOrder, createOfflineOrder as universalCreateOfflineOrder, getOfflineOrderPreview as universalGetOfflineOrderPreview, getPricePreview as universalGetPricePreview, managementGetOrder as universalManagementGetOrder, managementListOrders as universalManagementListOrders, markAsPaid as universalMarkAsPaid, memberGetOrder as universalMemberGetOrder, memberListOrders as universalMemberListOrders, pauseOrder as universalPauseOrder, postponeEndDate as universalPostponeEndDate, requestCancellation as universalRequestCancellation, resumeOrder as universalResumeOrder, } from './pricing-plans-v2-order-orders.universal';
export const __metadata = { PACKAGE_NAME: '@wix/pricing-plans' };
export function memberGetOrder(httpClient) {
    return (_id, options) => universalMemberGetOrder(_id, options, 
    // @ts-ignore
    { httpClient });
}
export function memberListOrders(httpClient) {
    return (options) => universalMemberListOrders(options, 
    // @ts-ignore
    { httpClient });
}
export function requestCancellation(httpClient) {
    return (_id, effectiveAt) => universalRequestCancellation(_id, effectiveAt, 
    // @ts-ignore
    { httpClient });
}
export function createOfflineOrder(httpClient) {
    return (planId, memberId, options) => universalCreateOfflineOrder(planId, memberId, options, 
    // @ts-ignore
    { httpClient });
}
export function getOfflineOrderPreview(httpClient) {
    return (planId, memberId, options) => universalGetOfflineOrderPreview(planId, memberId, options, 
    // @ts-ignore
    { httpClient });
}
export function getPricePreview(httpClient) {
    return (planId, options) => universalGetPricePreview(planId, options, 
    // @ts-ignore
    { httpClient });
}
export function managementGetOrder(httpClient) {
    return (_id, options) => universalManagementGetOrder(_id, options, 
    // @ts-ignore
    { httpClient });
}
export function managementListOrders(httpClient) {
    return (options) => universalManagementListOrders(options, 
    // @ts-ignore
    { httpClient });
}
export function postponeEndDate(httpClient) {
    return (_id, endDate) => universalPostponeEndDate(_id, endDate, 
    // @ts-ignore
    { httpClient });
}
export function cancelOrder(httpClient) {
    return (_id, effectiveAt) => universalCancelOrder(_id, effectiveAt, 
    // @ts-ignore
    { httpClient });
}
export function markAsPaid(httpClient) {
    return (_id) => universalMarkAsPaid(_id, 
    // @ts-ignore
    { httpClient });
}
export function pauseOrder(httpClient) {
    return (_id) => universalPauseOrder(_id, 
    // @ts-ignore
    { httpClient });
}
export function resumeOrder(httpClient) {
    return (_id) => universalResumeOrder(_id, 
    // @ts-ignore
    { httpClient });
}
export const onOrderCanceled = EventDefinition('wix.pricing_plans.v2.order_canceled', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderCreated = EventDefinition('wix.pricing_plans.v2.order_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.startDate' },
            { path: 'entity.endDate' },
            { path: 'entity.earliestEndDate' },
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.cancellation.requestedDate' },
            { path: 'entity.pausePeriods.pauseDate' },
            { path: 'entity.pausePeriods.resumeDate' },
            { path: 'entity.currentCycle.startedDate' },
            { path: 'entity.currentCycle.endedDate' },
            { path: 'entity.cycles.startedDate' },
            { path: 'entity.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderUpdated = EventDefinition('wix.pricing_plans.v2.order_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.startDate' },
            { path: 'entity.endDate' },
            { path: 'entity.earliestEndDate' },
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.cancellation.requestedDate' },
            { path: 'entity.pausePeriods.pauseDate' },
            { path: 'entity.pausePeriods.resumeDate' },
            { path: 'entity.currentCycle.startedDate' },
            { path: 'entity.currentCycle.endedDate' },
            { path: 'entity.cycles.startedDate' },
            { path: 'entity.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderStartDateChanged = EventDefinition('wix.pricing_plans.v2.order_start_date_changed', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderPurchased = EventDefinition('wix.pricing_plans.v2.order_purchased', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderStarted = EventDefinition('wix.pricing_plans.v2.order_started', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderCycleStarted = EventDefinition('wix.pricing_plans.v2.order_cycle_started', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderAutoRenewCanceled = EventDefinition('wix.pricing_plans.v2.order_auto_renew_canceled', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderEnded = EventDefinition('wix.pricing_plans.v2.order_ended', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderEndDatePostponed = EventDefinition('wix.pricing_plans.v2.order_end_date_postponed', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderMarkedAsPaid = EventDefinition('wix.pricing_plans.v2.order_marked_as_paid', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderPaused = EventDefinition('wix.pricing_plans.v2.order_paused', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onOrderResumed = EventDefinition('wix.pricing_plans.v2.order_resumed', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.startDate' },
            { path: 'data.order.endDate' },
            { path: 'data.order.earliestEndDate' },
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.cancellation.requestedDate' },
            { path: 'data.order.pausePeriods.pauseDate' },
            { path: 'data.order.pausePeriods.resumeDate' },
            { path: 'data.order.currentCycle.startedDate' },
            { path: 'data.order.currentCycle.endedDate' },
            { path: 'data.order.cycles.startedDate' },
            { path: 'data.order.cycles.endedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export { CancellationCause, CancellationEffectiveAt, OrderMethod, OrderStatus, OrderType, PaymentStatus, PeriodUnit, ReasonNotSuspendable, Set, SortOrder, Status, WebhookIdentityType, } from './pricing-plans-v2-order-orders.universal';
