// @wix/pricing-plans_orders/build/es/src/pricing-plans-v2-order-orders.meta.js
import * as ambassadorWixPricingPlansV2Order from './pricing-plans-v2-order-orders.http';
export function memberGetOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.memberGetOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/member/orders/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function memberListOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Order.memberListOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/member/orders',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function requestCancellation() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.requestCancellation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/member/orders/{id}/cancel',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createOfflineOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Order.createOfflineOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/checkout/orders/offline',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getOfflineOrderPreview() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Order.getOfflineOrderPreview(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/checkout/orders/preview-offline',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getPricePreview() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Order.getPricePreview(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/checkout/orders/price-preview',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function managementGetOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.managementGetOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/orders/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function managementListOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixPricingPlansV2Order.managementListOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/orders',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function postponeEndDate() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.postponeEndDate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/orders/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function cancelOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.cancelOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/orders/{id}/cancel',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function markAsPaid() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.markAsPaid(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/orders/{id}/mark-as-paid',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function pauseOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.pauseOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/orders/{id}/pause',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function resumeOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixPricingPlansV2Order.resumeOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/orders/{id}/resume',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
