// @wix/pricing-plans_orders/build/es/src/pricing-plans-v2-order-orders.http.js
import { toURLSearchParams } from '@wix/metro-runtime';
import { transformSDKTimestampToRESTTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/metro-runtime';
function resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl(opts) {
    const domainToMappings = {
        'www._base_domain_': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'forums.wixapps.net': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
            {
                srcPath: '/pricing-plans/v2/checkout',
                destPath: '/v2/checkout',
            },
            {
                srcPath: '/paid-plans/v1/orders',
                destPath: '/v1/orders',
            },
        ],
        '*.wixapps.net': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        '*.pub.wix-code.com': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
            {
                srcPath: '/_api/pricing-plans',
                destPath: '',
            },
        ],
        'forums._base_domain_': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'editor-flow.wixapps.net': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'editor.wixapps.net': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/pricing-plans/v2/checkout',
                destPath: '/v2/checkout',
            },
            {
                srcPath: '/pricing-plans/v2/orders',
                destPath: '/v2/orders',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/membership-api',
                destPath: '',
            },
            {
                srcPath: '/paid-plans',
                destPath: '',
            },
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressMembershipV2OrdersCheckoutServiceUrl(opts) {
    const domainToMappings = {
        'www._base_domain_': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'forums.wixapps.net': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
            {
                srcPath: '/pricing-plans/v2/checkout',
                destPath: '/v2/checkout',
            },
            {
                srcPath: '/paid-plans/v1/orders',
                destPath: '/v1/orders',
            },
        ],
        '*.wixapps.net': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        '*.pub.wix-code.com': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
            {
                srcPath: '/_api/pricing-plans',
                destPath: '',
            },
        ],
        'forums._base_domain_': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'editor-flow.wixapps.net': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'editor.wixapps.net': [
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/pricing-plans/v2/checkout',
                destPath: '/v2/checkout',
            },
            {
                srcPath: '/pricing-plans/v2/orders',
                destPath: '/v2/orders',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/membership-api',
                destPath: '',
            },
            {
                srcPath: '/paid-plans',
                destPath: '',
            },
            {
                srcPath: '/_api/paid-plans',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl(opts) {
    const domainToMappings = {
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/pricing-plans-member-orders',
                destPath: '',
            },
            {
                srcPath: '/_api/pricing-plans/v2/member',
                destPath: '/v2/member',
            },
        ],
        '*.pub.wix-code.com': [
            {
                srcPath: '/_api/pricing-plans-member-orders',
                destPath: '',
            },
            {
                srcPath: '/_api/pricing-plans/v2/member',
                destPath: '/v2/member',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/pricing-plans/v2/member',
                destPath: '/v2/member',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/pricing-plans/v2/member',
                destPath: '/v2/member',
            },
        ],
        _: [
            {
                srcPath: '/_api/pricing-plans/v2/member',
                destPath: '/v2/member',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/pricing-plans/v2/member',
                destPath: '/v2/member',
            },
        ],
        'editor.wixapps.net': [
            {
                srcPath: '/_api/pricing-plans/v2/member',
                destPath: '/v2/member',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/pricing-plans_orders';
/**
 * Gets an order by ID for the currently logged-in member.
 *
 * The `memberGetOrder()` function returns a Promise that resolves to information about a specified order for the currently logged-in member.
 */
export function memberGetOrder(payload) {
    function __memberGetOrder({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'GET',
            methodFqn: 'com.wixpress.membership.v2.orders.member.MemberOrdersService.GetOrder',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl({
                protoPath: '/v2/member/orders/{id}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'order.startDate' },
                        { path: 'order.endDate' },
                        { path: 'order.earliestEndDate' },
                        { path: 'order.createdDate' },
                        { path: 'order.updatedDate' },
                        { path: 'order.cancellation.requestedDate' },
                        { path: 'order.pausePeriods.pauseDate' },
                        { path: 'order.pausePeriods.resumeDate' },
                        { path: 'order.currentCycle.startedDate' },
                        { path: 'order.currentCycle.endedDate' },
                        { path: 'order.cycles.startedDate' },
                        { path: 'order.cycles.endedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __memberGetOrder;
}
/**
 * Lists orders for the currently logged-in member.
 *
 * The `memberListOrders()` function returns a Promise that resolves to a list of up to 100 pricing plan orders.
 */
export function memberListOrders(payload) {
    function __memberListOrders({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'GET',
            methodFqn: 'com.wixpress.membership.v2.orders.member.MemberOrdersService.ListOrders',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl({
                protoPath: '/v2/member/orders',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orders.startDate' },
                        { path: 'orders.endDate' },
                        { path: 'orders.earliestEndDate' },
                        { path: 'orders.createdDate' },
                        { path: 'orders.updatedDate' },
                        { path: 'orders.cancellation.requestedDate' },
                        { path: 'orders.pausePeriods.pauseDate' },
                        { path: 'orders.pausePeriods.resumeDate' },
                        { path: 'orders.currentCycle.startedDate' },
                        { path: 'orders.currentCycle.endedDate' },
                        { path: 'orders.cycles.startedDate' },
                        { path: 'orders.cycles.endedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __memberListOrders;
}
/**
 * Starts the process of canceling an order.
 *
 *
 * The `requestCancellation()` function returns a Promise that resolves when the order cancellation is successfully requested.
 *
 * For orders with recurring payments, a cancellation can be set to occur either immediately or at the next payment date. For orders with one-time payments, a cancellation occurs immediately after the request is processed.
 *
 * Requesting an order cancellation starts the cancellation process. There may be some operations that continue to be processed before the status of the order is changed to `"CANCELED"`. For example, payments might need to be refunded before the order is fully canceled.
 *
 * #### Canceling during the free trial period.
 *
 * When a buyer cancels their order during the free trial period, the buyer's subscription expires at the end of the free trial period and they won't be billed. The buyer may continue using the benefits until the end of the free trial period.
 *
 */
export function requestCancellation(payload) {
    function __requestCancellation({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.orders.member.MemberOrdersService.RequestCancellation',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl({
                protoPath: '/v2/member/orders/{id}/cancel',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __requestCancellation;
}
/**
 * Creates an order for a buyer who purchased the plan with an offline transaction.
 *
 * The `createOfflineOrder()` function returns a Promise that resolves to an `order` object when the order has been created.
 *
 * Payment of an offline order is handled in 1 of 2 ways.
 * - When creating the order, select `true` in the `paid` request parameter.
 * - After creation, with the [`markAsPaid()`](#markaspaid) function.
 *
 * When creating a non-free offline order:
 * - The order's status is set to `"PENDING"` if the start date is in the future. Otherwise, the status is set to `"ACTIVE"`.
 * The order's last payment status is set to `"UNPAID"` or `"PAID"`.
 *
 * When creating a free offline order:
 * - The order's status is set to `"PENDING"` if the start date is in the future. Otherwise, the status is set to `"ACTIVE"`.
 * - The order's last payment status is set to `"NOT_APPLICABLE"`.
 */
export function createOfflineOrder(payload) {
    function __createOfflineOrder({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [{ path: 'startDate' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.orders.CheckoutService.CreateOfflineOrder',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
                protoPath: '/v2/checkout/orders/offline',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'order.startDate' },
                        { path: 'order.endDate' },
                        { path: 'order.earliestEndDate' },
                        { path: 'order.createdDate' },
                        { path: 'order.updatedDate' },
                        { path: 'order.cancellation.requestedDate' },
                        { path: 'order.pausePeriods.pauseDate' },
                        { path: 'order.pausePeriods.resumeDate' },
                        { path: 'order.currentCycle.startedDate' },
                        { path: 'order.currentCycle.endedDate' },
                        { path: 'order.cycles.startedDate' },
                        { path: 'order.cycles.endedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createOfflineOrder;
}
/**
 * Provides a preview of an offline order as if it was purchased.
 *
 * The `getOfflineOrderPreview()` function returns a Promise that resolves to a temporary preview of the offline order.
 *
 * The preview uses the same logic as purchasing a plan, but the preview is not saved. Because an order is not actually created, the preview's `_id` and `subscriptionId` properties are displayed as a string of multiple zero characters (`000000-0000`).
 *
 * If [taxes are configured](https://support.wix.com/en/article/pricing-plans-setting-up-tax-collection) for the site, taxes are applied to the preview. If not, `tax` previews as `null`.
 *
 * You can preview the order to check purchase limitations, but the limitations are not enforced for the preview. If a pricing plan has a limit on the amount of purchases per buyer, that limit is not considered for generating the preview. But, if that limit has been reached and this order would then exceed the amount of purchases permitted for this buyer, then `purchaseLimitExceeded` will return as `true`. Thus function is not available to the buyer. You specify the member ID for the buyer whose order should be previewed. To get a general price preview for a plan that's not buyer-specific, use the [`getPricePreview()`](#getpricepreview) function.
 */
export function getOfflineOrderPreview(payload) {
    function __getOfflineOrderPreview({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [{ path: 'startDate' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.orders.CheckoutService.GetOfflineOrderPreview',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
                protoPath: '/v2/checkout/orders/preview-offline',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'order.startDate' },
                        { path: 'order.endDate' },
                        { path: 'order.earliestEndDate' },
                        { path: 'order.createdDate' },
                        { path: 'order.updatedDate' },
                        { path: 'order.cancellation.requestedDate' },
                        { path: 'order.pausePeriods.pauseDate' },
                        { path: 'order.pausePeriods.resumeDate' },
                        { path: 'order.currentCycle.startedDate' },
                        { path: 'order.currentCycle.endedDate' },
                        { path: 'order.cycles.startedDate' },
                        { path: 'order.cycles.endedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getOfflineOrderPreview;
}
/**
 * Retrieves a preview of an order's pricing as if it was purchased.
 *
 * The `getPricePreview()` function returns a Promise that resolves to a temporary preview of the order's price.
 *
 * The price preview uses the same logic for calculating prices as used when purchasing a plan, but the preview is not saved. If [taxes are configured](https://support.wix.com/en/article/pricing-plans-setting-up-tax-collection) for the site, taxes are applied to the preview. If not, the `tax` previews as `null`.
 *
 * Buyers do not have to be logged in to preview the price, as such, the details returned by this function are not buyer-specific. To generate a preview of a purchase for a specific-buyer, use the [`getOfflineOrderPreview()`](#getofflineorderpreview).
 */
export function getPricePreview(payload) {
    function __getPricePreview({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.orders.CheckoutService.GetPricePreview',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
                protoPath: '/v2/checkout/orders/price-preview',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __getPricePreview;
}
/**
 * Retrieves an order by ID.
 *
 * The `managementGetOrder()` function returns a Promise that resolves to information about the specified order.
 */
export function managementGetOrder(payload) {
    function __managementGetOrder({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'GET',
            methodFqn: 'com.wixpress.membership.v2.orders.OrderManagementService.GetOrder',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
                protoPath: '/v2/orders/{id}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'order.startDate' },
                        { path: 'order.endDate' },
                        { path: 'order.earliestEndDate' },
                        { path: 'order.createdDate' },
                        { path: 'order.updatedDate' },
                        { path: 'order.cancellation.requestedDate' },
                        { path: 'order.pausePeriods.pauseDate' },
                        { path: 'order.pausePeriods.resumeDate' },
                        { path: 'order.currentCycle.startedDate' },
                        { path: 'order.currentCycle.endedDate' },
                        { path: 'order.cycles.startedDate' },
                        { path: 'order.cycles.endedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __managementGetOrder;
}
/**
 * Lists pricing plan orders.
 *
 * The `managementListOrders()` function returns a Promise that resolves to a list of up to 50 pricing plan orders. You can specify options for filtering, sorting, and paginating the results.
 *
 * This function returns the orders on the site. To list orders for the currently logged-in member, use [`memberListOrders()`](#memberlistorders).
 */
export function managementListOrders(payload) {
    function __managementListOrders({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'GET',
            methodFqn: 'com.wixpress.membership.v2.orders.OrderManagementService.ListOrders',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
                protoPath: '/v2/orders',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orders.startDate' },
                        { path: 'orders.endDate' },
                        { path: 'orders.earliestEndDate' },
                        { path: 'orders.createdDate' },
                        { path: 'orders.updatedDate' },
                        { path: 'orders.cancellation.requestedDate' },
                        { path: 'orders.pausePeriods.pauseDate' },
                        { path: 'orders.pausePeriods.resumeDate' },
                        { path: 'orders.currentCycle.startedDate' },
                        { path: 'orders.currentCycle.endedDate' },
                        { path: 'orders.cycles.startedDate' },
                        { path: 'orders.cycles.endedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __managementListOrders;
}
/**
 * Extends the duration of a pricing plan order by postponing the order's `endDate`.
 *
 * The `postponeEndDate()` function returns a Promise that resolves when the order's end date is successfully changed.
 *
 * The new end date and time must be later than the order's current `endDate`.
 *
 * Postponing the end date of an order does not impact payments. For example, if the pricing plan is for a membership to an online lecture series, and you want to extend the duration of the series because the lecturer could not attend some sessions, you can postpone the end date of the orders for all relevant participants. The participants will not be billed additionally.
 *
 * Postponing an order causes the following changes:
 * - The `endDate` for the order is adjusted to the new end date.
 */
export function postponeEndDate(payload) {
    function __postponeEndDate({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [{ path: 'endDate' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'PATCH',
            methodFqn: 'com.wixpress.membership.v2.orders.OrderManagementService.PostponeEndDate',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
                protoPath: '/v2/orders/{id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
        };
        return metadata;
    }
    return __postponeEndDate;
}
/**
 * Cancels an existing order.
 *
 * The `cancelOrder()` function returns a Promise that resolves when the order is successfully canceled.
 *
 * For orders with recurring payments, a cancellation can be set to occur either `IMMEDIATELY` or at the `NEXT_PAYMENT_DATE`.
 * For orders with one-time payments, a cancellation occurs `IMMEDIATELY`.
 *
 * Canceling an order changes the order status to `CANCELED`.
 *
 * #### Canceling during the free trial period.
 *
 * When a site owner cancels an ordered plan during the free trial period, they choose to apply the cancellation `IMMEDIATELY` or at the `NEXT_PAYMENT_DATE`.
 *
 * Canceling `IMMEDIATELY` will end the subscription for the buyer
 * immediately, even during the free trial period and the buyer won't be billed.
 *
 * Canceling at the `NEXT_PAYMENT_DATE` allows the buyer to continue using the benefits of the subscription until the end of the free trial period. Then, the subscription ends and the buyer is not billed.
 */
export function cancelOrder(payload) {
    function __cancelOrder({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.orders.OrderManagementService.CancelOrder',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
                protoPath: '/v2/orders/{id}/cancel',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __cancelOrder;
}
/**
 * Marks an offline order as paid.
 *
 * The `markAsPaid()` function returns a Promise that resolves when the offline order is successfully marked as paid.
 *
 * The entire order is marked as paid, even if the order's payments are recurring.
 *
 * >**Note:** Marking separate payment cycles as paid is not yet supported. Subsequent offline payments do trigger events and emails, but are not registered as additional offline payments.
 *
 * Marking an offline order as paid causes the following changes:
 * - The order's `lastPaymentStatus` changes to `"PAID"`.
 * - The order's status changes to either `"PENDING"` or `"ACTIVE"`, depending on the order's `startDate`.
 *
 * An error occurs if you attempt to:
 * - Mark an already-paid, offline order as paid. You cannot make an offline order as paid twice.
 * - Mark an online order as paid. The `markAsPaid()` function is supported for offline orders only.
 */
export function markAsPaid(payload) {
    function __markAsPaid({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.orders.OrderManagementService.MarkAsPaid',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
                protoPath: '/v2/orders/{id}/mark-as-paid',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __markAsPaid;
}
/**
 * Pauses a pricing plan order.
 *
 * The `pauseOrder()` function returns a Promise that resolves when the order is successfully paused.
 *
 * For orders with recurring payments, `pauseOrder()` also pauses the payment schedule. Buyers are not charged when an order is paused. Use `pauseOrder()`, for example, if the buyer is away and would like to put their pricing plan membership on hold until they return. Pausing an order affects the end date of the order by adding the time the order is paused to the `endDate`. You can only pause orders with an `"ACTIVE`" status.
 *
 * Pausing an order causes the following changes:
 * - The order status changes to `"PAUSED"`.
 * - The `pausePeriods` array is updated.
 *
 * The `endDate` and the `earliestEndDate` for the order are adjusted to include the pause period when the order is resumed.
 *
 * Paused orders can be continued with the [`resumeOrder()`](#resumeorder) function.
 */
export function pauseOrder(payload) {
    function __pauseOrder({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.orders.OrderManagementService.PauseOrder',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
                protoPath: '/v2/orders/{id}/pause',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __pauseOrder;
}
/**
 * Resumes a paused pricing plan order.
 *
 * The `resumeOrder()` function returns a Promise that resolves when a paused order is successfully resumed.
 *
 * For orders with recurring payments, `resumeOrder()` also restarts the payment schedule.
 *
 * Resuming an order causes the following changes:
 * - The order status changes to `"ACTIVE"`.
 * - The `pausePeriods` array is updated.
 * - The `endDate` for the order is adjusted to include the pause period.
 * - The `earliestEndDate` is adjusted to include the pause period. (This property is reserved for future use).
 */
export function resumeOrder(payload) {
    function __resumeOrder({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.order',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.orders.OrderManagementService.ResumeOrder',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
                protoPath: '/v2/orders/{id}/resume',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __resumeOrder;
}
