// @wix/identity_verification/build/es/src/iam-verification-v1-start-response-verification.meta.js
import * as ambassadorWixIamVerificationV1StartResponse from './iam-verification-v1-start-response-verification.http';
export function start() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamVerificationV1StartResponse.start(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/Start',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function verifyDuringAuthentication() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamVerificationV1StartResponse.verifyDuringAuthentication(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/auth/verify',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function resendDuringAuthentication() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamVerificationV1StartResponse.resendDuringAuthentication(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/auth/resend',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
