// @wix/sdk-runtime/cjs/build/transformations/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fieldMask = exports.timestamp = exports.duration = exports.bytes = exports.float = exports.pageUrlV2 = exports.pageUrl = exports.document = exports.video = exports.videoV2 = exports.image = exports.address = exports.audio = void 0;
exports.audio = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/audio',
        name: 'transformSDKAudioToRESTAudio',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/audio',
        name: 'transformRESTAudioToSDKAudio',
    },
};
exports.address = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/address',
        name: 'transformSDKAddressToRESTAddress',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/address',
        name: 'transformRESTAddressToSDKAddress',
    },
};
exports.image = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/image',
        name: 'transformSDKImageToRESTImage',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/image',
        name: 'transformRESTImageToSDKImage',
    },
};
exports.videoV2 = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/video-v2',
        name: 'transformSDKVideoV2ToRESTVideoV2',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/video-v2',
        name: 'transformRESTVideoV2ToSDKVideoV2',
    },
};
exports.video = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/video',
        name: 'transformSDKVideoToRESTVideo',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/video',
        name: 'transformRESTVideoToSDKVideo',
    },
};
exports.document = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/document',
        name: 'transformSDKDocumentToRESTDocument',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/document',
        name: 'transformRESTDocumentToSDKDocument',
    },
};
exports.pageUrl = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/page-url',
        name: 'transformSDKPageURLToRESTPageURL',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/page-url',
        name: 'transformRESTPageURLToSDKPageURL',
    },
};
exports.pageUrlV2 = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/page-url-v2',
        name: 'transformSDKPageURLV2ToRESTPageURLV2',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/page-url-v2',
        name: 'transformRESTPageURLV2ToSDKPageURLV2',
    },
};
exports.float = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/float',
        name: 'transformSDKFloatToRESTFloat',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/float',
        name: 'transformRESTFloatToSDKFloat',
    },
};
exports.bytes = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/bytes',
        name: 'transformSDKBytesToRESTBytes',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/bytes',
        name: 'transformRESTBytesToSDKBytes',
    },
};
exports.duration = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/duration',
        name: 'transformSDKDurationToRESTDuration',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/duration',
        name: 'transformRESTDurationToSDKDuration',
    },
};
exports.timestamp = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/timestamp',
        name: 'transformSDKTimestampToRESTTimestamp',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/timestamp',
        name: 'transformRESTTimestampToSDKTimestamp',
    },
};
exports.fieldMask = {
    toREST: {
        source: '@wix/sdk-runtime/transformations/field-mask',
        name: 'transformSDKFieldMaskToRESTFieldMask',
    },
    fromREST: {
        source: '@wix/sdk-runtime/transformations/field-mask',
        name: 'transformRESTFieldMaskToSDKFieldMask',
    },
};
