// @wix/identity_recovery/build/cjs/src/iam-recovery-v1-recovery-recovery.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.recover = exports.sendActivationEmail = exports.sendRecoveryEmail = void 0;
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const metro_runtime_1 = require("@wix/metro-runtime");
function resolveWixIamRecoveryV1RecoveryServiceUrl(opts) {
    const domainToMappings = {
        _: [
            {
                srcPath: '/_iam/recovery',
                destPath: '',
            },
            {
                srcPath: '/_api/iam/recovery',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/_api/iam/recovery',
                destPath: '',
            },
        ],
    };
    return (0, metro_runtime_1.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
/**
 * Sends a member an email containing a customized link to a Wix-managed page
 * where the member can set a new password for their account.
 */
function sendRecoveryEmail(payload) {
    function __sendRecoveryEmail({ host }) {
        const metadata = {
            entityFqdn: 'wix.iam.recovery.v1.recovery',
            method: 'POST',
            methodFqn: 'wix.iam.recovery.v1.RecoveryService.SendRecoveryEmail',
            url: resolveWixIamRecoveryV1RecoveryServiceUrl({
                protoPath: '/v1/send-email',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __sendRecoveryEmail;
}
exports.sendRecoveryEmail = sendRecoveryEmail;
/**
 * Sends an activation email with an activation token
 * making the transition from initial contact to a site member
 */
function sendActivationEmail(payload) {
    function __sendActivationEmail({ host }) {
        const metadata = {
            entityFqdn: 'wix.iam.recovery.v1.recovery',
            method: 'POST',
            methodFqn: 'wix.iam.recovery.v1.RecoveryService.SendActivationEmail',
            url: resolveWixIamRecoveryV1RecoveryServiceUrl({
                protoPath: '/v1/activation-email',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __sendActivationEmail;
}
exports.sendActivationEmail = sendActivationEmail;
function recover(payload) {
    function __recover({ host }) {
        const metadata = {
            entityFqdn: 'wix.iam.recovery.v1.recovery',
            method: 'POST',
            methodFqn: 'wix.iam.recovery.v1.RecoveryService.Recover',
            url: resolveWixIamRecoveryV1RecoveryServiceUrl({
                protoPath: '/v1/recover',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'identity.createdDate' },
                        { path: 'identity.updatedDate' },
                        { path: 'identity.identityProfile.customFields.value.dateValue' },
                        { path: 'additionalData.dateValue' },
                    ],
                },
                {
                    transformFn: float_1.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'identity.identityProfile.customFields.value.numValue' },
                        { path: 'additionalData.numValue' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __recover;
}
exports.recover = recover;
